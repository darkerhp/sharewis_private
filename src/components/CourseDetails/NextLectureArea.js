import React, { PropTypes } from 'react';
import ReactNative from 'react-native';
import I18n from 'react-native-i18n';

import BaseStyles from '../../lib/baseStyles';
import Lecture from '../../modules/models/Lecture';

const {
  View,
  StyleSheet,
  Text,
  Image,
  TouchableOpacity,
} = ReactNative;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  courseImageWrapper: {
    flex: 2,
    overflow: 'hidden',
    alignItems: 'flex-start',
    justifyContent: 'center',
    maxWidth: BaseStyles.deviceWidth - 20, // 親コンポーネントのpadding分引いた画面幅
  },
  courseImage: {},
  nextLectureTextWrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: BaseStyles.navBarBackgroundColor,
  },
  nextLectureText: {
    color: 'white',
    fontWeight: '900',
  },
  nextLectureTitleText: {
    color: 'white',
    position: 'absolute',
    top: 30,
    left: 75,
    backgroundColor: 'rgba(0,0,0,0)',
  },

});

const videoImageSrc = require('./images/video.png');
const quizImageSrc = require('./images/quiz.png');
const textImageSrc = require('./images/text.png');

const getNextLectureImageSrc = (nextLecture) => {
  switch (nextLecture.type) {
    case Lecture.TYPE_VIDEO:
      return videoImageSrc;
    case Lecture.TYPE_QUIZ:
      return quizImageSrc;
    case Lecture.TYPE_TEXT:
      return textImageSrc;
    default:
      return videoImageSrc;
  }
};

const NextLectureArea = ({ nextLecture, handlePressNextLecture, containerStyle }) =>
  <View style={[styles.container, containerStyle]}>
    <TouchableOpacity onPress={handlePressNextLecture}>
      <View style={styles.courseImageWrapper}>
        <Image
          source={getNextLectureImageSrc(nextLecture)}
          style={styles.courseImage}
          resizeMode={Image.resizeMode.contain}
        />
        <Text style={styles.nextLectureTitleText}>{nextLecture.title}</Text>
      </View>
      <View style={styles.nextLectureTextWrapper}>
        <Text style={styles.nextLectureText}>{I18n.t('nextLecture')}</Text>
      </View>
    </TouchableOpacity>
  </View>;

NextLectureArea.propTypes = {
  nextLecture: PropTypes.shape({}).isRequired,
  handlePressNextLecture: PropTypes.func.isRequired,
  containerStyle: PropTypes.any, // eslint-disable-line
};

export default NextLectureArea;
