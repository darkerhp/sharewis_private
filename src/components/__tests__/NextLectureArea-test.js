/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import NextLectureArea from '../CourseDetails/NextLectureArea';

jest.mock('bugsnag-react-native', () => 'Bugsnag');
jest.mock('react-native-i18n', () => ({ t: () => '' }));

describe('NextLectureArea', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <NextLectureArea
          containerStyleId={0}
          handlePressNextLecture={() => null}
          nextLecture={{
            type: 'lecture'
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
