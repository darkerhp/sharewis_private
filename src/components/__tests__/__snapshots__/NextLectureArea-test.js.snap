exports[`NextLectureArea renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "justifyContent": "center"
      },
      0
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 2,
          "justifyContent": "center",
          "overflow": "hidden"
        }
      }>
      <Image
        resizeMode="contain"
        source={1}
        style={Object {}} />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "color": "white",
            "left": 75,
            "position": "absolute",
            "top": 30
          }
        } />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#579eff",
          "flex": 1,
          "justifyContent": "center"
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "900"
          }
        }>
        [missing \"en.nextLecture\" translation]
      </Text>
    </View>
  </View>
</View>
`;
